### ISSUE TITLE
[Feature] Delete old accounts

### ISSUE THREAD

**Issue by matunixe** [tootsuite/mastodon#3811]
Hello there,

I don't know if it's actually possible, I searched but found nothing. So here it is:
It would be very useful if we can have the ability to delete (eg. with a rake task) old accounts. Like account who haven't logged in for 1 month.

In our instance we have **many, many** sleeping accounts who have been abandoned by their owner.

Thanks in advance,

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by Kimik0** [comment]
I'm not sure a mere month of absence should be enough to delete an account. Maybe 3 months, or sooner if they hadn't tooted at all or something.

It's a good idea in general though to automate it somewhat. Mastodon's user counts are kinda inflated by people who have checked it out once then decided it's not (yet) for them. Plus all the username squatting back when people were worried about 'owning' their username everywhere and the number of instances still made that seem possible.

**Comment by matunixe** [comment]
Yes, the value I said (1 month) is not mandatory. 3 months seems very reasonable from here.

**Comment by joyeusenoelle** [comment]
Maybe a task that sends out an email to all accounts that haven't been logged in for 90 days, and then another that deletes all accounts that were sent the email more than a week ago? (In other words, there's a seven-day window between sending the email and deleting the account.) That gives people a chance to say "whoa, wait, I forgot about that" but gets rid of genuinely abandoned accounts.

**Comment by matunixe** [comment]
Thanks everyone to  give interesting ideas! 💜

**Comment by trwnh** [comment]
Going to put in a voice of dissent here: while it may feel nice to clear out hard drive space you perceive is unnecessary, a *lot* more consideration needs to be given to the time limits, and even the general idea.

I've seen people come back to Mastodon after a year or more; perhaps they joined in April 2017 like many others, and they felt alienated by Twitter's recent changes (which are perennial these days). 1 month or 3 months is definitely too short; t-shirt enough to qualify as perhaps a social media vacation rather than a permanent absence.

The other consideration is that many of these people signed up under a user agreement that did not mention that their account would be deleted for inactivity at all. It's a bit of a chicken/egg scenario, but ideally they would have to agree to updated terms. Of course, this is ultimately up to admins to administer their instances however they want, so...

That then means that any such inactivity limit would have to be confgurable and *clearly stated at time of sign up*. Since this represents a drastic change away from expected norms (no other service provider deletes, say, old email accounts -- except for the one time when Yahoo did), it needs to be communicated clearly to users.

And perhaps one final consideration to what happens to usernames of these accounts -- are they deleted as well? Are they permanently reserved? Are they freed up as well? What happens if someone comes back after a certain limit and wants to sign up again on the same instance?
